crypto/dkg_feldmanvss.go

```diff
@@ -287,7 +280,23 @@ func (s *feldmanVSSstate) receiveShare(origin index, data []byte) {
 		return
 	}
 
+	// at this point, tag the private message as received
+	s.xReceived = true
```

Comment:
The DKG layer does not force the calling layer to deduplicate private messages (or broadcasts) and handles duplicates itself. Because of the networking constraints you described, receiving duplicates does not lead to disqualifications from Joint-Feldman. 
The `DKGProcessor` has two levels of punishment: 
  - `Disqualify` excludes the participant from joint-feldman, and asks the upper layer to slash the participant from the protocol. Duplicate messages or being offline does not trigger this function. 
  - `FlagMisbehaviour` just flags that a participant is not acting 100% right but no serious actions will/can be taken against it. In the case of duplicate messages, the upper layer may just decide to rate limit the incoming messages, or reduce the participant's reputation score in the networking layer.

Reply:
Btw, this has been the case since the first implementation of DKG. What is new is that if the first private message sent is invalid, the receiver would not offer the sender a second chance to fix it. The first message would be considered the main private message, whether it is valid or invalid.

Reply:
Makes sense, thank you. I was thinking "misbehaviour" is worse than it actually is 😄